mlflow/models/signature.py

```diff
@@ -101,6 +101,7 @@ def __init__(
             self.outputs = outputs
         self.params = params
         self.__is_signature_from_type_hint = False
+        self.__type_hint_from_example = False
```

Comment:
nit: Do we want to preserve consistency with `__is_signature_from_type_hint` syntax with `__is_type_hint_from_example` since it's also a Bool?